test/compilable/issue17167.sh

```diff
@@ -16,8 +18,6 @@ src="$bin_base.d"
 echo 'void main() {}' > "${src}"
 
 # Only compile, not link, since optlink can't handle long file names
-$DMD -m"${MODEL}" "${DFLAGS}" -c -of"${bin}" "${src}" || exit 1
+$DMD -m"${MODEL}" "${DFLAGS}" -c -of"${bin}" "${src}"
 
 rm -rf "${dir:?}"/"$TEST_NAME"
-
-echo Success
```

Comment:
We no longer need to write sth. to the output file to create it as it's now created by default and only removed in the error case. Of course, that behavior could be easily changed in the `sh_do_test` script, but the point is that this is superfluous, was often forgotten in the past and should be handled by the wrapper.